pkg/schedule/schedulers/balance_range.go

```diff
@@ -617,88 +624,65 @@ func (s *balanceRangeScheduler) prepare(cluster sche.SchedulerCluster, opInfluen
 				count += cluster.GetStoreLearnerCountByRange(source.GetID(), kr.StartKey, kr.EndKey)
 			}
 		}
-		scoreMap[source.GetID()] = int64(count)
-		totalScore += int64(count)
+		scoreMap[source.GetID()] = float64(count)
+		totalScore += float64(count)
 	}
 
 	sort.Slice(sources, func(i, j int) bool {
 		rule := job.Rule
-		iop := opInfluence.GetStoreInfluence(sources[i].GetID()).GetStoreInfluenceByRole(rule)
-		jop := opInfluence.GetStoreInfluence(sources[j].GetID()).GetStoreInfluenceByRole(rule)
+		iop := float64(opInfluence.GetStoreInfluence(sources[i].GetID()).GetStoreInfluenceByRole(rule))
+		jop := float64(opInfluence.GetStoreInfluence(sources[j].GetID()).GetStoreInfluenceByRole(rule))
 		iScore := scoreMap[sources[i].GetID()]
 		jScore := scoreMap[sources[j].GetID()]
 		return iScore+iop > jScore+jop
 	})
 
-	averageScore := int64(0)
-	averageScore = totalScore / int64(len(sources))
+	averageScore := float64(0)
+	averageScore = totalScore / float64(len(sources))
 
 	tolerantSizeRatio := int64(float64(totalScore) * adjustRatio)
 	if tolerantSizeRatio < 1 {
 		tolerantSizeRatio = 1
 	}
+	solver.tolerantSource = tolerantSizeRatio
 	return &balanceRangeSchedulerPlan{
 		SchedulerCluster: cluster,
 		stores:           sources,
 		scoreMap:         scoreMap,
-		source:           nil,
-		target:           nil,
-		region:           nil,
 		averageScore:     averageScore,
 		job:              job,
-		opInfluence:      opInfluence,
-		tolerate:         tolerantSizeRatio,
+		solver:           solver,
 	}, nil
 }
 
-func (p *balanceRangeSchedulerPlan) sourceStoreID() uint64 {
-	return p.source.GetID()
-}
-
-func (p *balanceRangeSchedulerPlan) targetStoreID() uint64 {
-	return p.target.GetID()
-}
-
-func (p *balanceRangeSchedulerPlan) score(storeID uint64) int64 {
+func (p *balanceRangeSchedulerPlan) score(storeID uint64) float64 {
 	return p.scoreMap[storeID]
 }
 
-func (p *balanceRangeSchedulerPlan) shouldBalance(scheduler string) bool {
-	sourceInfluence := p.opInfluence.GetStoreInfluence(p.sourceStoreID())
-	sourceInf := sourceInfluence.GetStoreInfluenceByRole(p.job.Rule)
-	// Sometimes, there are many remove-peer operators in the source store, we don't want to pick this store as source.
-	if sourceInf < 0 {
-		sourceInf = -sourceInf
+func (p *balanceRangeSchedulerPlan) sourceScore(storeID uint64) float64 {
+	originScore := p.score(storeID)
+	s := p.solver
+	tolerantResource := s.getTolerantResource()
+	// to avoid schedule too much, if A's core greater than B and C a little
+	// we want that A should be moved out one region not two
+	influence := s.getOpInfluence(storeID)
+	if influence > 0 {
+		influence = -influence
 	}
```

Comment:
Do we need to add some comments about why reverse influence

Reply:
done